feat: Add options to extract label and value from data in mapToDropdownOptions

fozarix0 opened the pull request:
- Allow nested labels and values to be extracted out for converting into dropdown items.

Pan-YS committed:
feat: Add mapLabel and mapValue options

Pan-YS committed:
docs: Add examples to mapToDropdownOptions

Pan-YS committed:
test: Add tests for mapToDropdownOptions

Pan-YS committed:
fix: Login button not dimming when loading

Somehow Tailwind CSS's '!important' doesn't work on classes here, so removed in this commit

fozarix0 commented:
I've also committed the fix for the button spinner not dimming when loading in the fourth (latest, as of now) commit.

I would also like to point out that the code may also be refactored to just include the .iwFormBtnDisabled in the class directly without the conditional, since the Tailwind CSS's 'disabled:' is already included in the original .viBtnDisabled class.
```
  class="iwFormSubmitBtn"
  :class="{ 'iwFormBtnDisabled': isLoading }"
```
to
```
  class="iwFormSubmitBtn iwFormBtnDisabled"
```

kenng commented on utils/IwFormFunctions.ts:
```diff
@@ -1,19 +1,37 @@
-    data: Record<string, string>,
-    params: IwFormSelectOptionParam = {}) {
+    data: Record<string, string | number | { [key: string]: any }>,
+    params: IwFormSelectOptionParam = {}): Array<IwFormInputSelectOption> {
```
@freshnewuniquename change to 
`Record<string, string | number | { [key: string]: any }>,` to just
`Record<string, any>`

Pan-YS committed:
refactor: Update the options to extract text values from data

Remove mapLabel function and replace with labelKey property

Remove mapValue function and replace with valueKey property

Pan-YS committed:
refactor: Reorder fields to ascending alphabetical order

Pan-YS committed:
docs: add examples for array data use case

Pan-YS committed:
test: add test for array use case

Pan-YS committed:
docs: cleanup examples documentation

Pan-YS committed:
docs: fix typo
